analyzere/base_resources.py

```diff
@@ -48,6 +52,35 @@ def __deepcopy__(self, memo):
             return copy.deepcopy(self.__wrapped__, memo)
         return Reference(self._href)
 
+    def __getattribute__(self, name):
+        # Check for class methods and attributes before intercepting Proxied
+        # methods and attributes.
+        if hasattr(Reference, name):
+            return super(Reference, self).__getattribute__(name)
+
+        attr = Proxy.__getattribute__(self, name)
+        # Intercept all proxied attributes and methods and attempt to update
+        # the Reference ._id and ._href attributes.
+
+        def update_ref(item):
+            try:
+                id_ = item.id
+                href_ = urljoin(analyzere.base_url, item._get_path(id_))
```

Comment:
Ahh, ok. Maybe that's what I was missing. I thought `Resource` was a subclass of `Reference`.